rust/src/page_table.rs

```diff
@@ -0,0 +1,57 @@
+use std::io::{Read, Seek, SeekFrom};
+
+use arrow2::datatypes::DataType;
+use byteorder::{LittleEndian, ReadBytesExt};
+
+#[derive(Debug, Clone)]
+pub struct PageInfo {
+    pub position: i64,
+    pub length: i64,
+}
+
+#[derive(Debug, Clone)]
+pub struct PageTable {
+    page_info_map: Vec<Vec<PageInfo>>,
+}
+
+impl PageTable {
+    pub fn make<R: Read + Seek>(
+        file: &mut R,
+        page_table_position: u64,
+        num_columns: usize,
+        num_batches: usize,
+    ) -> PageTable {
+        // ARROW_ASSIGN_OR_RAISE(
+        //     auto buf, in->ReadAt(page_table_position, (num_columns * num_batches * 2 * sizeof(int64_t))));
+        //
+        // auto arr = ::arrow::Int64Array(num_columns * num_batches * 2, buf);
+        file.seek(SeekFrom::Start(page_table_position)).unwrap();
+        // let mut buf = vec![0u8; num_columns * num_batches * 2 * size_of::<i64>()];
+        // file.read_exact(&mut buf).unwrap();
+        // let buffer = arrow2::buffer::Buffer::from(buf);
+
+        let mut vec = vec![0i64; num_columns * num_batches * 2];
+        file.read_i64_into::<LittleEndian>(&mut vec).unwrap(); //TODO is it right?
+        let buffer = arrow2::buffer::Buffer::from(vec);
```

Comment:
Is this zero copy?

Reply:
I think so, but here is the [source code](https://github.com/jorgecarleitao/arrow2/blob/main/src/buffer/immutable.rs#L181) in case I may have some misunderstanding.